### PR TITLE
Changed RSS to Atom feed and improve paging

### PR DESCRIPTION
**Pull request by tcitworld**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes :warning: 
| Deprecations? | no
| Tests pass?   | yes
| Documentation | why not
| Translation   | no
| Fixed tickets | 
| License       | MIT

- [x] Use Atom instead of RSS
- [x] Paginate properly (well, not perfectly)
- [x] Improve a bit the tests with the new elements
- [x] Handle redirection from previous feed

Will have to rebase on https://github.com/wallabag/wallabag/pull/3207

### CODE REVIEW

**Comment by nicosomb** [comment]
BC breaks, yes. So, which version to target?

**Comment by tcitworld** [comment]
> BC breaks, yes.

Well, no app relies anymore on RSS afaik, so it's not really a BC break. Should be fine for 2.3.

**Comment by nicosomb** [comment]
> so it's not really a BC break

Maybe some users use these feeds in their RSS readers.

**Comment by tcitworld** [comment]
Well, Atom is supported by nearly all RSS Readers, so this won't be an issue. And I added back compatibility for old URLs, that we can remove in later major versions.

**Comment by nicosomb** [comment]
What about "Improve a bit the tests with the new elements" in your first post?

**Comment by tcitworld** [comment]
You're welcomed to do it, but it doesn't matter much from my PoV. Maybe just a add check on tags. \#todo

**Comment by j0k3r** [comment]
Why switching from rss to atom?

**Comment by tcitworld** [comment]
See for instance https://www.reddit.com/r/changelog/comments/428vdq/upcoming_reddit_change_switching_from_rss_20_to/
Also, a nice read http://nullprogram.com/blog/2013/09/23/

File: src/Wallabag/CoreBundle/Controller/RssController.php [hunk 1]
```diff
@@ -18,40 +17,52 @@ class RssController extends Controller
     /**
      * Shows unread entries for current user.
      *
-     * @Route("/{username}/{token}/unread.xml", name="unread_rss", defaults={"_format"="xml"})
+     * @Route("/feed/{username}/{token}/unread/{page}", name="unread_rss", defaults={"page": 1})
+     * @Route("/{username}/{token}/unread.xml", defaults={"page": 1})
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/RssController.php]: Don't remove the `_format`, it send the proper content type.
Otherwise, send the proper content type your self (in `showEntries`)

File: src/Wallabag/CoreBundle/Resources/views/themes/common/Entry/entries.xml.twig [hunk 2]
```diff
@@ -1,34 +1,46 @@
...
-                <guid>{{ entry.url }}</guid>
-                <pubDate>{{ entry.createdAt|date('D, d M Y H:i:s') }}</pubDate>
-                <description>
-                    <![CDATA[{%- if entry.readingTime > 0 -%}{{ 'entry.list.reading_time_minutes'|trans({'%readingTime%': entry.readingTime}) }}{%- else -%}{{ 'entry.list.reading_time_less_one_minute'|trans|raw }}{%- endif %}{{ entry.content|raw -}}]]>
-                </description>
-            </item>
-
+<feed xmlns="http://www.w3.org/2005/Atom">
+    <title>wallabag — {{type}} feed</title>
+    <subtitle type="html">RSS feed for {{ type }} entries</subtitle>
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/themes/common/Entry/entries.xml.twig]: Better _Atom feed_

File: src/Wallabag/CoreBundle/Resources/views/themes/common/Entry/entries.xml.twig [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
+    {% if entries | length > 0 %}
+        <updated>{{ (entries | first).createdAt | date('c') }}</updated> {# Indicates the last time the feed was modified in a significant way. #}
+    {% endif %}
+    <id>wallabag:{{ domainName | removeScheme | removeWww }}:{{ user }}:{{ type }}</id>
+    <link rel="alternate" type="text/html" href="{{ url(type) }}"/>
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/themes/common/Entry/entries.xml.twig]: I'm wondering it `url(` generates an absolute url or not.
Otherwise use `{{ absolute_url(path(type)) }}`

**Comment by j0k3r** [comment]
It's a BC break if we remove the previous urls.
Otherwise we can add them now and remove it in a futur version.

Good for 2.3 for me 👍

**Comment by tcitworld** [comment]
Rebased.

File: src/Wallabag/CoreBundle/Controller/FeedController.php [hunk 4]
```diff
@@ -12,82 +12,93 @@
...
+    public function showArchiveFeedAction(User $user, $page)
     {
-        return $this->showEntries('archive', $user, $request->query->get('page', 1));
+        return $this->showEntries('archive', $user, $page);
     }
 
     /**
      * Shows starred entries for current user.
      *
-     * @Route("/{username}/{token}/starred.xml", name="starred_rss", defaults={"_format"="xml"})
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/FeedController.php]: Let the default _format everywhere.

File: src/Wallabag/CoreBundle/Resources/translations/messages.da.yml [hunk 5]
```diff
@@ -53,7 +53,7 @@ config:
     page_title: 'Opsætning'
     tab_menu:
         settings: 'Indstillinger'
-        rss: 'RSS'
+        feed: 'RSS'
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/translations/messages.da.yml]: Shouldn't this be translated to Feed instead of RSS then?

File: tests/Wallabag/CoreBundle/Controller/FeedControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+        $config = $user->getConfig();
+        $config->setFeedToken('SUPERTOKEN');
+        $config->setFeedLimit(1);
+        $em->persist($config);
+        $em->flush();
+
+        $client = $this->getClient();
+
+        $client->request('GET', '/feed/admin/SUPERTOKEN/unread');
+        $this->assertEquals(200, $client->getResponse()->getStatusCode());
```
Comment (j0k3r) [inline tests/Wallabag/CoreBundle/Controller/FeedControllerTest.php]: use `assertSame` instead

File: src/Wallabag/UserBundle/Repository/UserRepository.php [hunk 7]
```diff
@@ -7,18 +7,18 @@
...
-     * @return User|null
+     * @return null|User
      */
-    public function findOneByUsernameAndRsstoken($username, $rssToken)
+    public function findOneByUsernameAndFeedtoken($username, $feedToken)
     {
         return $this->createQueryBuilder('u')
             ->leftJoin('u.config', 'c')
-            ->where('c.rssToken = :rss_token')->setParameter('rss_token', $rssToken)
+            ->where('c.feedToken = :feed_token')->setParameter('feed_token', $feedToken)
```
Comment (j0k3r) [inline src/Wallabag/UserBundle/Repository/UserRepository.php]: No DB migration? 🤔

**Comment by j0k3r** [comment]
Rebase needed.

**Comment by ewoks** [comment]
@tcitworld is there any chance you can take a look and rebase it soon? thanks in advance

**Comment by j0k3r** [comment]
poke @tcitworld ☝️

File: src/Wallabag/CoreBundle/Command/InstallCommand.php [hunk 8]
```diff
@@ -254,7 +254,7 @@ protected function setupAdmin()
         $question->setHidden(true);
         $user->setPlainPassword($this->io->askQuestion($question));
 
-        $user->setEmail($this->io->ask('Email', ''));
+        $user->setEmail($this->io->ask('Email', 'wallabag@wallabag.io'));
```
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Command/InstallCommand.php]: Why is this here?

Comment (j0k3r) [reply src/Wallabag/CoreBundle/Command/InstallCommand.php]: Sorry, didn't explain.
It's because of this: https://github.com/symfony/symfony/blob/3.4/src/Symfony/Component/Console/Helper/SymfonyQuestionHelper.php#L46

We can't define an ask question without a default value which isn't empty (as we used before).

Comment (tcitworld) [reply src/Wallabag/CoreBundle/Command/InstallCommand.php]: Okay, but still odd for this change to be in the PR.

Comment (j0k3r) [reply src/Wallabag/CoreBundle/Command/InstallCommand.php]: I agree but it was broken the tests.
